File: Algorithms/Bubble Sort/bubble_sort.py
# Bubble Sort

def bubble_sort(num_list):
	"""A sorting algorithm that returns a list of number in ascending order.

	Args:
		num_list (INT): a list of integers

	Returns:
		num_list: A sorted list of integers in ascending order
	"""	
	for items in range(len(num_list)-1,0,-1):
		for item in range(items):
			if num_list[item] > num_list[item+1]:
				# Swap
				temp = num_list[item]
				num_list[item] = num_list[item+1]
				num_list[item+1] = temp
	return num_list
	
	
print (bubble_sort([54,26,93,17,77,31,44,55,20]))

File: Algorithms/Bubble Sort/bubble-sort.ts
const sort = (list: number[]): number[] => {
  const temp = list;
  for (let i = 0; i < list.length; i++) {
    for (let j = 0; j < list.length - i - 1; j++) {
      if (temp[j] > temp[j + 1]) {
        const t = temp[j];
        temp[j] = temp[j + 1];
        temp[j + 1] = t;
      }
    }
  }
  return temp;
};

console.log(sort([4, 5, 1, 8, 9, 0]));
